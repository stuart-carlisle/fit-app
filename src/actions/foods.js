import { database } from '../firebase/firebase'

//SET_FOODS_PUBLIC
export const setFoodsPublic = (foodsPublic) => ({
    type:'SET_FOODS_PUBLIC',
    foodsPublic
  })


//SET_FOODS_PRIVATE
export const setFoodsPrivate = (foodsPrivate) => ({
  type:'SET_FOODS_PRIVATE',
  foodsPrivate
})

export const startSetFoodsPrivate = () => {
  return (dispatch,getState) => {
    const uid = getState().auth.uid
    return database.ref(`users/${uid}/foods`).once('value').then((snapshot)=>{
        const foodsPrivate = []
        snapshot.forEach((childSnapshot)=>{
          foodsPrivate.push({
            id:childSnapshot.key,
            ...childSnapshot.val()
          })
        })
        dispatch(setFoodsPrivate(foodsPrivate))
    })  
  }
}

//SET_FOODS_DIARY
export const setFoodsDiary = (foods) => ({
  type:'SET_FOODS_DIARY',
  foods
})

export const startSetFoodsDiary = () => {
  return (dispatch,getState) => {
    const uid = getState().auth.uid
    return database.ref(`users/${uid}/diary/foods`).once('value').then((snapshot)=>{
        const foods = []
        snapshot.forEach((childSnapshot)=>{
          foods.push({
            id:childSnapshot.key,
            ...childSnapshot.val()
          })
        })
        dispatch(setFoodsDiary(foods))
    })  
  }
}

// ADD_FOOD PRIVATE
export const addFoodPrivate = (food) => ({
  type: 'ADD_FOOD_PRIVATE',
  food
});

export const startAddFoodPrivate = (foodData) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid
    const { description, serving, nutrition, lastUsed } = foodData
    const food = { description, serving, nutrition, lastUsed }
   
    return database.ref(`users/${uid}/foods`).push(food).then((ref)=>{
      dispatch(addFoodPrivate({
        id:ref.key,
        ...food
      }))
    })
  }
}

// ADD_FOOD_DIARY
export const addFoodDiary = (food) => ({
  type: 'ADD_FOOD_DIARY',
  food
});

export const startAddFoodDiary = (foodData) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid
    const {description, serving, nutrition, diaryDate, numberOfServings, meal} = foodData
    const food = { description, serving, nutrition, diaryDate, numberOfServings, meal}
   
    return database.ref(`users/${uid}/diary/foods`).push(food).then((ref)=>{
      dispatch(addFoodDiary({
        id: ref.key,
        ...food
      }))
    })
  }
}

// UPDATE_FOOD_PRIVATE
export const updateFoodPrivate = (id, updates) => ({
  type: 'UPDATE_FOOD_PRIVATE',
  id,
  updates
});

export const startUpdateFoodPrivate = (id, updates) => {
  return (dispatch,getState) => {
    const uid = getState().auth.uid
    return database.ref(`users/${uid}/foods/${id}`).update(updates).then(()=>{
      dispatch(updateFoodPrivate(id,updates))
    })
  }
}

// REMOVE_FOOD_DIARY
export const removeFoodDiary = ({ id } = {}) => ({
  type: 'REMOVE_FOOD_DIARY',
  id
});


export const startRemoveFoodDiary = ({ id } = {}) => {
  return (dispatch,getState) => {
    const uid = getState().auth.uid
    return database.ref(`users/${uid}/diary/foods/${id}`).remove().then(()=>{
      dispatch(removeFoodDiary({id}))
    })
  }
}

// UPDATE_FOOD_DIARY
export const updateFoodDiary = (id, updates) => ({
  type: 'UPDATE_FOOD_DIARY',
  id,
  updates
});

export const startUpdateFoodDiary = (id, updates) => {
  return (dispatch,getState) => {
  const uid = getState().auth.uid
  return database.ref(`users/${uid}/diary/foods/${id}`).update(updates).then(()=>{
    dispatch(updateFoodDiary(id,updates))
  })
}}

// UPDATE_TARGETS
export const updateTargets = (updates) => ({
  type: 'UPDATE_TARGETS',
  updates
})

export const startUpdateTargets = (updates) => {
  return (dispatch,getState) => {
  const uid = getState().auth.uid
  return database.ref(`users/${uid}/diary/targets`).update(updates).then(()=>{
    dispatch(updateTargets(updates))
  })
}}

// SET_TARGETS
export const setTargets = (targets) => ({
  type:'SET_TARGETS',
  targets
})

export const startSetTargets = () => {
  return (dispatch,getState) => {
    const uid = getState().auth.uid
    return database.ref(`users/${uid}/diary/targets`).once('value').then((snapshot)=>{
      const targets = snapshot.val()
      if(!!targets){
      dispatch(setTargets(targets))
      }
  })  
}
}

// UPDATE_WEIGHT
export const updateWeight = (weight) => ({
  type: 'UPDATE_WEIGHT',
  weight
})

export const startUpdateWeight = (weight,date) => {
  return (dispatch,getState) => {
  const uid = getState().auth.uid
  return database.ref(`users/${uid}/diary/weights/${date}`).set({weight})
  .then(()=>{
    dispatch(updateWeight({weight: weight, date: date}))
  })
}}

//SET_WEIGHTS_DIARY
export const setWeightsDiary = (weights) => ({
  type:'SET_WEIGHTS_DIARY',
  weights
})

export const startSetWeightsDiary = () => {
  return (dispatch,getState) => {
    const uid = getState().auth.uid
    return database.ref(`users/${uid}/diary/weights`).once('value').then((snapshot)=>{
        const weights = []
        snapshot.forEach((childSnapshot)=>{
          weights.push({
            ...childSnapshot.val(),
            date:childSnapshot.key
          })
        })
        dispatch(setWeightsDiary(weights))
    })  
  }
}