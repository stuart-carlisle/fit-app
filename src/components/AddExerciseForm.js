import React, { useState , useEffect} from 'react'
import moment from 'moment'
import { connect } from 'react-redux'
import numeral from 'numeral'

export const AddExerciseForm = ({
  filters = {},
  onSubmit, 
  exercise = {
    description:'',
    type:'',
    energy:'',
    time:''
  }
}) => {
    const defaultEnergy = numeral(exercise.energy).value()
    const defaultTime = numeral(exercise.time).value()
    const date = filters.date
    const referenceDatabase = filters.referenceDatabase
    const [time, setTime] = useState(defaultTime)
    const [error, setError] = useState('')
    const [energy, setEnergy] = useState(defaultEnergy)
    const diaryDate = date ? date.valueOf() : moment().startOf('day').valueOf()
    const lastUsed = moment().valueOf()

    useEffect(()=>{
        
      let refreshed = JSON.parse(localStorage.getItem('refreshed'))
      if(refreshed){
        window.history.back()
        window.scrollTo(0,0)
        localStorage.removeItem('refreshed')
      }else{
        localStorage.setItem('refreshed',JSON.stringify(true))
      }
    },[])

    const onFormSubmit = (e) => {
        e.preventDefault();
        const exerciseData = { 
          description: exercise.description,
          type: exercise.type
        }
        if (numeral(time).value()<=0) {
          setError('Please provide a time greater than 0')
          window.scrollTo(0,0)
        } else {
          setError('')
          onSubmit(
            referenceDatabase,
            diaryDate,
            lastUsed,
            exerciseData,
            time.toString(),
            energy.toString()
          )
          localStorage.removeItem('refreshed')
        }
      }
    
    const onTimeChange = (e)=>{
        const time = e.target.value
          if (!time || time.match(/^\d{1,}(\.\d{0,1})?$/)) {
          setTime(time)
          const timeValue = numeral(time).value()
          setEnergy(Math.round(defaultEnergy * (timeValue/defaultTime)))
        }
    }

    const onEnergyChange = ()=>{
      
    }

    return (
        <div>
        <div className="content-container__background">
        <div className="content-container__form__update">
            <form className= "form" onSubmit={onFormSubmit}>
                {error && <p id="error-message" className="form__error">{error}</p>}
                  <h1 className="page-header__title-update"> {exercise.description} </h1>
                    <span className="text-input__span-container-form">
                      <div className="content-container__update-titles">
                        <h3 className="form__heading__update">Exercise Time (mins):</h3>
                        <h3 className="form__heading__update">Energy (calories): </h3>
                      </div>
                      <div className="content-container__update">
                        <input  //Exercise Duration
                          className="text-input__update-form"
                          type="text"
                          value={time}
                          onChange={onTimeChange}
                          placeholder="Enter a Duration"
                        />  
                        <input  //Energy Used
                          className="text-input__update-form"
                          type="text"
                          value={energy?energy:''}
                          onChange={onEnergyChange}
                          placeholder="Energy Used"
                        /> 
                      </div>
                    </span>
                <button className="button button--form">Add Exercise</button>
            </form>
          </div>
        </div>
      </div>  
    )
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  }
}


export default connect(mapStateToProps)(AddExerciseForm);