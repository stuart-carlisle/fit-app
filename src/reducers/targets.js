// targets Reducer

const targetsReducerDefaultState = {
  dailyEnergyTarget: '2400',
  dailyCarbsTarget: '400',
  dailyFatTarget: '70'
}

export default (state = targetsReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_DAILY_ENERGY_TARGET':
      return{
        ...state,
        dailyEnergyTarget: action.dailyEnergyTarget
      }
    case 'UPDATE_TARGETS':
      return{
        ...state,
        ...action.updates
      }
    case 'SET_TARGETS':
      return action.targets
    default:
      return state;
  }
};
