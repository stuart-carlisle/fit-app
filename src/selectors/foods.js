// Get visible expenses

export default (foods, { text, sortBy }) => {

  return foods.filter((food) => {
    const textMatch = food.description.toLowerCase().includes(text.toLowerCase())
    return textMatch;
  }).sort((a, b) => {
    if (sortBy === 'date') {
      return a.lastUsed < b.lastUsed ? 1 : -1;
    } else if (sortBy === 'alphabetical') {
      return a.description > b.description ? 1 : -1;
    }
  })
} 
