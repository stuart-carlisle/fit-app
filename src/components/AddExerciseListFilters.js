import React, { useState } from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByDate, sortByAlphabetical, setReferenceDatabaseToPublic } from '../actions/filters';
import { setExercisesPublic } from '../actions/exercises'
import algoliasearch from 'algoliasearch/lite'

export const AddExerciseListFilters = ({
    filters,
    setTextFilter, 
    setReferenceDatabasePublic,
    sortByDate, 
    sortByAlphabetical,
    setExercisesPublic 
  }) => {
 
  const searchClient = algoliasearch (
      process.env.ALGOLIA_APP_ID,
      process.env.ALGOLIA_API_KEY
  )

  const index = searchClient.initIndex('exercises');

  const [text, setText] = useState('')
    
  const onTextChange = (e) => {
    setTextFilter(e.target.value)
    setText(e.target.value)
  }

  const onSortChange = (e) => {
    if (e.target.value === 'date') {
      sortByDate()
    } else if (e.target.value === 'alphabetical') {
      sortByAlphabetical()
    }
  }

  const onSubmit = (e) => {
    e.preventDefault()
    index
      .search({
        query: text
      })
      .then((responses) => {
        const array = responses.hits
        const newExercises = array.map((exercise)=>{
          return{
            description:exercise.description,
            type:exercise.type,
            energy:exercise.energy,
            time:exercise.time,
            id:exercise.objectID
          }
        })
        setExercisesPublic(newExercises)
      }).then(()=>{
        setReferenceDatabasePublic()
      })
}

    return (
      
      <div className="header__content--filters"> 
      <form className="filters-text-input__span-container" onSubmit={onSubmit}>
          <div className="input-group__item-page-header">
            <input
              type="text"
              className="filters-text-input"
              placeholder="SEARCH EXERCISES"
              value={text}
              onChange={onTextChange}
            />
          </div>
          <div className="input-group__item-page-header">
            <select
              className="minimal"
              value={filters.sortBy}
              onChange={onSortChange}
            >
              <option value="date">DATE</option>
              <option value="alphabetical">ALPHABETICAL</option>
            </select>
          </div>
            <button className="button filters-button--form-page-header">Search Online</button>
            </form>   
        </div>
    )
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  };
};

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAlphabetical: () => dispatch(sortByAlphabetical()),
  setReferenceDatabasePublic: () => dispatch(setReferenceDatabaseToPublic()),
  setExercisesPublic: (exercises) => dispatch(setExercisesPublic(exercises))
})

export default connect(mapStateToProps,mapDispatchToProps)(AddExerciseListFilters);

