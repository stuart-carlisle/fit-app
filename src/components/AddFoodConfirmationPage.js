import React from 'react'
import { connect } from 'react-redux'
import AddFoodForm from './AddFoodForm'
import { Link } from 'react-router-dom'
import { startAddFoodDiary, startAddFoodPrivate, startUpdateFoodPrivate } from '../actions/foods'

export const AddFoodConfirmationPage = (props) => {
  
  const onSubmit = (referenceDatabase, diaryDate, lastUsed, numberOfServings, food, meal) => {
    if (referenceDatabase==='public'){
      props.startAddFoodPrivate({...food,lastUsed})
    } else if (referenceDatabase==='private'){
      props.startUpdateFoodPrivate(props.food.id,{...food,lastUsed})
    }
    props.startAddFoodDiary({...food, diaryDate, numberOfServings, meal})
    props.history.push('/diary');
  }
  

  return (
     <div>
     <Link className="button__back-button" to={`/add`}><img className="image__header" src="/images/back-arrow.svg"/></Link>
        <div className="page-header__filter page-header__filter--with-back">
          <div className="content-container">
            <h1 className="page-header__title">ADD FOOD</h1>
          </div>
        </div>
        <div className="show-for-mobile page-header__margin"></div>
        <div className="content-container">
        <AddFoodForm id="add-food-form" targets={props.targets} food={props.food} onSubmit={onSubmit} />
        </div>
     </div>
 )
}

const mapStateToProps = (state,props) => {
  if(state.filters.referenceDatabase==='public'){
    return {
      food: state.foodsPublic.find((food) => food.id === props.match.params.id),
      targets: state.targets
    }
  }else if(state.filters.referenceDatabase==='private'){
    return {
      food: state.foodsPrivate.find((food) => food.id === props.match.params.id),
      targets: state.targets
    }
  }
}

const mapDispatchToProps = (dispatch) =>({
    startAddFoodDiary: (food) => dispatch(startAddFoodDiary(food)),
    startUpdateFoodPrivate: (id, food) => dispatch(startUpdateFoodPrivate(id, food)),
    startAddFoodPrivate: (food) => dispatch(startAddFoodPrivate(food))
})

export default connect(mapStateToProps, mapDispatchToProps)(AddFoodConfirmationPage);




