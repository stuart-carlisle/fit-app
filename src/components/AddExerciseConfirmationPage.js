import React from 'react'
import { connect } from 'react-redux'
import AddExerciseForm from './AddExerciseForm'
import { Link } from 'react-router-dom'
import { startAddExerciseDiary, startAddExercisePrivate, startUpdateExercisePrivate } from '../actions/exercises'

export const AddExerciseConfirmationPage = (props) => {
  
  const onSubmit = (referenceDatabase, diaryDate, lastUsed, exerciseDescAndType, time, energy) => {
    if (referenceDatabase==='public'){
      props.startAddExercisePrivate({...exerciseDescAndType, time, energy, lastUsed})
    } else if (referenceDatabase==='private'){
      props.startUpdateExercisePrivate(props.exercise.id,{...exerciseDescAndType, time, energy, lastUsed})
    }
    props.startAddExerciseDiary({...exerciseDescAndType, time, diaryDate, energy})
    props.history.push('/diary');
  }
  

  return (
     <div>
     <Link className="button__back-button" to={`/add-exercise`}><img className="image__header" src="/images/back-arrow.svg"/></Link>
        <div className="page-header__filter page-header__filter--with-back">
          <div className="content-container">
            <h1 className="page-header__title">ADD EXERCISE</h1>
          </div>
        </div>
        <div className="show-for-mobile page-header__margin"></div>
        <div className="content-container">
        <AddExerciseForm id="add-exercise-form" exercise={props.exercise} onSubmit={onSubmit} />
        </div>
     </div>
 )
}

const mapStateToProps = (state,props) => {
  if(state.filters.referenceDatabase==='public'){
    return {
      exercise: state.exercisesPublic.find((exercise) => exercise.id === props.match.params.id)
    }
  }else if(state.filters.referenceDatabase==='private'){
    return {
      exercise: state.exercisesPrivate.find((exercise) => exercise.id === props.match.params.id)
    }
  }
}

const mapDispatchToProps = (dispatch) =>({
    startAddExerciseDiary: (exercise) => dispatch(startAddExerciseDiary(exercise)),
    startUpdateExercisePrivate: (id, exercise) => dispatch(startUpdateExercisePrivate(id, exercise)),
    startAddExercisePrivate: (exercise) => dispatch(startAddExercisePrivate(exercise))
})

export default connect(mapStateToProps, mapDispatchToProps)(AddExerciseConfirmationPage);




