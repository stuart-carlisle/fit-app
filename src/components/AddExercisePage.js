import React, { useEffect } from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import AddExerciseListFilters from './AddExerciseListFilters'
import AddExercisesList from './AddExercisesList'
import { setTextFilter, sortByDate, setReferenceDatabaseToPrivate } from '../actions/filters'

export const AddExercisePage = ({ setTextFilter, sortByDate, setReferenceDatabaseToPrivate}) => {
  useEffect(()=>{
    localStorage.removeItem('refreshed')
      setTextFilter('')
      sortByDate()
      setReferenceDatabaseToPrivate()
  },[])
    return(
      <div>
      <Link className="button__back-button" to={`/diary`}><img className="image__header" src="/images/back-arrow.svg"/></Link>
    <div className="page-header__filter page-header__filter--with-back">
      <div className="content-container">
        <AddExerciseListFilters />
      </div>  
    </div>
    <AddExercisesList />
    <div className="content-container">
    <Link className="button button--link" to={`/create-exercise`}>+ CREATE CUSTOM EXERCISE</Link>
    </div>
    </div>
    )
}

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  setReferenceDatabaseToPrivate: () => dispatch(setReferenceDatabaseToPrivate())
})

export default connect (undefined, mapDispatchToProps)(AddExercisePage)