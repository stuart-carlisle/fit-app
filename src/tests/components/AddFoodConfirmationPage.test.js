import React from 'react'
import { shallow } from 'enzyme'
import { AddFoodConfirmationPage   } from '../../components/AddFoodConfirmationPage'
import foods from '../fixtures/foods'
import targets from '../fixtures/targets'
import moment from 'moment'

test('Should render AddFoodConfirmationPage with props correctly',()=>{
    const startAddFoodPrivate = jest.fn()
    const startUpdateFoodPrivate = jest.fn()
    const startAddFoodDiary = jest.fn()
    const history = { push: jest.fn() }
    const wrapper = shallow(
        <AddFoodConfirmationPage 
            startAddFoodDiary={startAddFoodDiary}
            startAddFoodPrivate={startAddFoodPrivate} 
            startUpdateFoodPrivate={startUpdateFoodPrivate}
            history={history}
            food={foods[1]}
            targets={targets}
        />
    )
    expect(wrapper).toMatchSnapshot()
})

test('Should handle add food from private foods',()=>{
    const filters =  {
        date:moment(0).startOf('day'),
        text: '',
        sortBy: 'date',
        referenceDatabase: 'private',
        chartType: {
            chart: 'calorie-tracker', 
            option: 'week'
        },
        meal: 'breakfast'
    }

    const startAddFoodPrivate = jest.fn()
    const startUpdateFoodPrivate = jest.fn()
    const startAddFoodDiary = jest.fn()
    const history = { push: jest.fn() }
    const wrapper = shallow(
        <AddFoodConfirmationPage 
            startAddFoodDiary={startAddFoodDiary}
            startAddFoodPrivate={startAddFoodPrivate} 
            startUpdateFoodPrivate={startUpdateFoodPrivate}
            history={history}
            food={foods[1]}
            targets={targets}
        />
    )
    wrapper.find('#add-food-form').prop('onSubmit')(filters.referenceDatabase, filters.date.valueOf(), moment().valueOf(), 1, {description:foods[1].description,nutrition:foods[1].nutrition,serving:foods[1].serving}, filters.meal)
    expect(startUpdateFoodPrivate).toHaveBeenLastCalledWith(
            foods[1].id,
            { 
                description: foods[1].description,
                nutrition: foods[1].nutrition,
                serving: foods[1].serving,
                lastUsed:moment().valueOf()
            }
             
        )
    expect(startAddFoodDiary).toHaveBeenLastCalledWith(
            { 
                description: foods[1].description,
                nutrition: foods[1].nutrition,
                serving: foods[1].serving,
                numberOfServings: 1,
                meal: filters.meal,
                diaryDate:filters.date.valueOf()
            }
    )
    expect(startAddFoodPrivate).not.toHaveBeenCalled()
    expect(history.push).toHaveBeenLastCalledWith('/diary')
})

test('Should handle add food from public foods',()=>{
    const filters = {
        date:moment(0).add(4, 'days').startOf('day'),
        text: '',
        sortBy: 'date',
        referenceDatabase: 'public',
        chartType: {
            chart: 'calorie-tracker', 
            option: 'week'
        },
        meal: 'lunch'
    }
    const startAddFoodPrivate = jest.fn()
    const startUpdateFoodPrivate = jest.fn()
    const startAddFoodDiary = jest.fn()
    const history = { push: jest.fn() }
    const wrapper = shallow(
        <AddFoodConfirmationPage 
            startAddFoodDiary={startAddFoodDiary}
            startAddFoodPrivate={startAddFoodPrivate} 
            startUpdateFoodPrivate={startUpdateFoodPrivate}
            history={history}
            food={foods[1]}
            targets={targets}
        />
    )
    wrapper.find('#add-food-form').prop('onSubmit')(filters.referenceDatabase, filters.date.valueOf(), moment().valueOf(), 1, {description:foods[1].description,nutrition:foods[1].nutrition,serving:foods[1].serving}, filters.meal)
    expect(startAddFoodPrivate).toHaveBeenLastCalledWith(
            { 
                description: foods[1].description,
                nutrition: foods[1].nutrition,
                serving: foods[1].serving,
                lastUsed:moment().valueOf()
            }
             
        )
    expect(startAddFoodDiary).toHaveBeenLastCalledWith(
            { 
                description: foods[1].description,
                nutrition: foods[1].nutrition,
                serving: foods[1].serving,
                numberOfServings: 1,
                meal: filters.meal,
                diaryDate:filters.date.valueOf()
            }
    )
    expect(startUpdateFoodPrivate).not.toHaveBeenCalled()
    expect(history.push).toHaveBeenLastCalledWith('/diary')
})
