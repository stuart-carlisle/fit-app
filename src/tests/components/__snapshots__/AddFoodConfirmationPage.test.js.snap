// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render AddFoodConfirmationPage with props correctly 1`] = `
<div>
  <Link
    className="button__back-button"
    replace={false}
    to="/add"
  >
    <img
      className="image__header"
      src="/images/back-arrow.svg"
    />
  </Link>
  <div
    className="page-header__filter page-header__filter--with-back"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header__title"
      >
        ADD FOOD
      </h1>
    </div>
  </div>
  <div
    className="show-for-mobile page-header__margin"
  />
  <div
    className="content-container"
  >
    <Connect(AddFoodForm)
      food={
        Object {
          "description": "CHEESE ON TOAST",
          "diaryDate": 0,
          "id": "2",
          "lastUsed": 0,
          "meal": "lunch",
          "numberOfServings": "1",
          "nutrition": Object {
            "energy": Object {
              "amount": "340",
            },
            "protein": Object {
              "amount": "2",
              "unit": "g",
            },
            "satFat": Object {
              "amount": "0.5",
              "unit": "g",
            },
            "sugars": Object {
              "amount": "3",
              "unit": "g",
            },
            "totalCarbs": Object {
              "amount": "34",
              "unit": "g",
            },
            "totalFat": Object {
              "amount": "1.2",
              "unit": "g",
            },
          },
          "serving": Object {
            "servingSize": Object {
              "size": "200",
              "unit": "g",
            },
            "servingsPerContainer": "1",
          },
        }
      }
      id="add-food-form"
      onSubmit={[Function]}
      targets={
        Object {
          "dailyCarbsTarget": "200",
          "dailyEnergyTarget": "2000",
          "dailyFatTarget": "4",
        }
      }
    />
  </div>
</div>
`;
